R/clean_redcap_long.R

```diff
@@ -22,19 +26,27 @@
 
 clean_redcap_long <- function(db_data_long,
                               db_metadata_long,
-                              linked_arms) {
+                              linked_arms,
+                              enable_mixed_structure = FALSE) {
   # Repeating Instrument Check ----
   # Check if database supplied contains any repeating instruments to map onto
   # `redcap_repeat_*` variables
 
   has_repeat_forms <- "redcap_repeat_instance" %in% names(db_data_long)
+  has_mixed_structure_forms <- FALSE
 
   # Apply checkmate checks
   assert_data_frame(db_data_long)
   assert_data_frame(db_metadata_long)
 
   if (has_repeat_forms) {
-    check_repeat_and_nonrepeat(db_data_long)
+    if (enable_mixed_structure) {
+      db_data_long <- convert_mixed_instrument(db_data_long, db_metadata_long)
+      mixed_structure_forms <- get_mixed_structure_fields(db_data_long)
```

Comment:
I'm a little confused about what calling this here after converting the data is doing. It looks like on the test data it's now flagging the nonrepeating fields?

Reply:
I have to update this anyway per the atomic comment above, but on the line below it was needed to update `has_mixed_structure_forms`. I wanted to still handle the case where a user allows/enables mixed structure on a database that has repeating forms, but none that are actually mixed-structure. 

In this case `convert_mixed_instrument` won't have any effect, and `mixed_structure_forms()` will still return all `FALSE` values, leaving `has_mixed_structure_forms` `FALSE` down the line.